transform/event_origin_index_provider.go

```diff
@@ -0,0 +1,34 @@
+package transform
+
+import (
+	"github.com/RoaringBitmap/roaring/roaring64"
+	"github.com/streamingfast/bstream/transform"
+	"github.com/streamingfast/dstore"
+)
+
+const EventOriginIndexShortName = "eventorigin"
+
+func NewEventOriginIndexProvider(
+	store dstore.Store,
+	possibleIndexSizes []uint64,
+	eventOrigins map[EventOrigin]bool,
+) *transform.GenericBlockIndexProvider {
+	return transform.NewGenericBlockIndexProvider(
+		store,
+		EventOriginIndexShortName,
+		possibleIndexSizes,
+		getEventOriginFilterFunc(eventOrigins),
+	)
+}
+
+func getEventOriginFilterFunc(EventOrigins map[EventOrigin]bool) func(transform.BitmapGetter) []uint64 {
```

Comment:
Any reason to capitalize the argument name? I think it should be `eventOrigins`.